### DISCUSSION TITLE
set environement variables to be used across multiple scheduled events

### DISCUSSION THREAD

**Discussion by mxdev88** [jhuckaby/Cronicle#582] [Q&A] [answered]
Version 0.9.17

Not an issue, a question (sorry i looked through old question but couldn't find my answers).

I would like to share environment variables across several scheduled jobs. Is there a way to achieve that (ideally using the web interface). For example, I would like to set variables in a given category that will be used for all scheduled events in that category? with the possibility to override at job level. Is that possible currently? If not, can it be implemented with a plugin?  

I've read you are preparing a new release so maybe something that will be possible in Orchestra?

**Comment by jhuckaby** [comment]
Moving this to a discussion.

**Comment by jhuckaby** [comment, marked as answer]
This is totally possible.  Just use a Plugin, and set Plugin parameters.  Those all become environment variables in jobs, and you can just assign all your jobs in your category to use that one Plugin.

Select "Hidden" to hide the environment variables from the event editor.

<img width="516" alt="Screenshot 2023-03-06 at 1 56 01 PM" src="https://user-images.githubusercontent.com/700765/223243991-27fcc85f-3f45-4f5f-b85f-b7780dce574a.png">

**Comment by mxdev88** [comment]
Ah great ! When you say "just use": Do you mean that this plugin already exists? Or I have to create it?

**Reply by jhuckaby** [reply]
What Plugin were you going to use before?  Just make a copy of it, and edit the parameters to add your desired environment variables.

For example, if you were going to use the Shell Plugin, just edit it, and click the "Copy Plugin..." button.  Then add your params to it.
